### PR TITLE
CNF-13448: Create resources needed by ClusterInstance

### PR DESCRIPTION
**Pull request by irinamihai**
Description:
- create the namespace, BMC secrets, pull secret and extra-manifests ConfigMaps
- updates for the new ClusterInstance CRD

ToDo:
- update testcases

/cc @Missxiaoguo @bartwensley

### CODE REVIEW

**Review by bartwensley — COMMENTED** [review]
Looks good

File: internal/controllers/utils/utils.go [hunk 1]
```diff
@@ -73,6 +73,69 @@ func ValidateJsonAgainstJsonSchema(schema string, input string) error {
...
 
+func GetBMCDetailsForClusterInstance(node map[string]interface{}, clusterRequest string) (
+	string, string, string, error) {
+	// Get the BMC details.
+	bmcCredentialsDetailsInterface, bmcCredentialsDetailsExist := node["bmcCredentialsDetails"]
+
+	oranUtilsLog.Info(
+		"[getBMCDetailsForClusterInstance]",
+		"bmcCredentialsDetailsExist ", fmt.Sprintf("%v", bmcCredentialsDetailsInterface),
+	)
```
[the remark this answers is not in the document]
Comment (irinamihai) [reply internal/controllers/utils/utils.go:84]: Yes, it was  just for dev purposes. I'll remove this.

File: internal/controllers/utils/utils.go [hunk 2]
```diff
@@ -73,6 +73,69 @@ func ValidateJsonAgainstJsonSchema(schema string, input string) error {
...
+		} else {
+			secretName =
+				extractBeforeDot(strings.ToLower(nodeHostnameInterface.(string))) +
+					"-bmc-secret"
+		}
+	} else {
+		secretName = bmcCredentialsNameInterface.(map[string]interface{})["name"].(string)
+	}
+
+	return username, password, secretName, nil
```
[the remark this answers is not in the document]
Comment (irinamihai) [reply internal/controllers/utils/utils.go:136]: I passed the username/pass in clear in the ClusterRequest and the final secret contains the username/pass in base64 format as that's how it's stored by default.

```
apiVersion: v1
data:
  password: YmJi
  username: YWFh
kind: Secret
metadata:
  creationTimestamp: "2024-07-09T16:15:38Z"
  name: site-sno-du-1-bmc-secret
  namespace: site-sno-du-1
  resourceVersion: "128748777"
  uid: ad6ec0ef-1496-4a79-9076-db3eade5bf43
type: Opaque
```
We can improve this if desired.

[the remark this answers is not in the document]
Comment (bartwensley) [reply internal/controllers/utils/utils.go:131]: Not urgent, but we should update this so the username/pass in the ClusterRequest are also in base64 format.

**Pull request merged**